src/Graphics/Widgets/BoundingBoxWidget.cc

```diff
@@ -26,87 +26,185 @@ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 {
-  center = center_;
-  right = right_;
-  down = down_;
-  in = in_;
+  center_ = trans.get_translation_point();
+  scaledEigvecs_ = trans.get_rotation_vectors();
+  getEigenValuesAndEigenVectors();
+  initWidgetCreation(idGenerator, widgetNum, widgetIter);
 }
 
 BoundingBoxWidget::BoundingBoxWidget(const Core::GeometryIDGenerator& idGenerator,
-  double scale, const BoxPosition& pos, const Point& origin, const BBox& bbox)
-  : WidgetBase(idGenerator, "BoundingBox", true, origin)
+                                     const std::string& name, double scale, const BoxPosition& pos,
+                                     const Point& center, int widgetNum, int widgetIter,
+                                     const BBox& bbox)
+  : CompositeWidget(idGenerator, name), scale_(scale), center_(center), bbox_(bbox), widgetsIndex_(-1)
 {
-  auto colorScheme(ColorScheme::COLOR_UNIFORM);
-  //get all the bbox edges
   Point c,r,d,b;
-  pos.getPosition(c,r,d,b);
-  auto x = r - c, y = d - c, z = b - c;
-  std::vector<Point> points = {
-    c + x + y + z,
-    c + x + y - z,
-    c + x - y + z,
-    c + x - y - z,
-    c - x + y + z,
-    c - x + y - z,
-    c - x - y + z,
-    c - x - y - z };
-  uint32_t point_indicies[] = {
-    0, 1, 0, 2, 0, 4,
-    7, 6, 7, 5, 3, 7,
-    4, 5, 4, 6, 1, 5,
-    3, 2, 3, 1, 2, 6
-  };
-  const auto num_strips = 50;
-  std::vector<Vector> tri_points;
-  std::vector<Vector> tri_normals;
-  std::vector<uint32_t> tri_indices;
-  std::vector<ColorRGB> colors;
-  GlyphGeom glyphs;
-  //generate triangles for the cylinders.
-  for (auto edge = 0; edge < 24; edge += 2)
-  {
-    glyphs.addCylinder(points[point_indicies[edge]], points[point_indicies[edge + 1]], scale, num_strips, ColorRGB(), ColorRGB());
-  }
-  //generate triangles for the spheres
-  for (const auto& a : points)
+  pos.getPosition(center_, r, d, b);
+  auto centerVector = Vector(center_);
+  scaledEigvecs_ = { Vector(r), Vector(d), Vector(b) };
+  for(auto& vec : scaledEigvecs_)
+    vec -= centerVector;
+
+  getEigenValuesAndEigenVectors();
+  initWidgetCreation(idGenerator, widgetNum, widgetIter);
+}
+
+void BoundingBoxWidget::createWidgets(const GeometryIDGenerator& idGenerator, int widgetNum,
+                                      int widgetIter)
+{
+  std::string boxName = widgetName(BoundingBoxWidgetSection::BOX, widgetNum, widgetIter);
+  widgets_.push_back(WidgetFactory::createBox(idGenerator, diagonalLength_ * scale_ * boxScale_,
+                                              center_, corners_, bbox_));
+  widgets_[++widgetsIndex_]->setToTranslate(MouseButton::LEFT);
+
+  for(int i = 0; i < 8; i++)
+    addCornerSphere(i, idGenerator, scaledEigvecs_[i/2], widgetNum, widgetIter);
+
+  for (int i = 0; i < 6; i++)
+    addFaceSphere(i, idGenerator, widgetNum, widgetIter);
+
+  glm::mat4 scaleTrans = glm::mat4(1.0);
+  for(int i = 0; i < 3; i++)
+    for(int j = 0; j < 3; j++)
+      scaleTrans[i][j] = eigvecs_[i][j];
+
+  for (int i = 0; i < 6; i++)
```

Comment:
Magic number --> named constant

Reply:
Yeah, the loop with eigvecs should change